### DISCUSSION TITLE
Can I Export Postgres DB from Vercel?

### DISCUSSION THREAD

**Discussion by sun-ks** [vercel/community#3863] [Help] [answered]
### Summary

I am going to export my Postgres to set up it on my local PC.  Is it possible?

Regards

### Steps to Reproduce

I am going to export my Postgres to set up it on my local PC.  Is it possible?

Regards

**Comment by leerob** [comment, marked as answer]
You can use a tool like `pg_dump` https://www.postgresql.org/docs/current/app-pgdump.html

**Comment by sun-ks** [comment]
Thanks a lot for helping.
Should I use SLI of Vercel or I can use a simulation of the terminal on web ? (it locaed on https://vercel.com/sun-ks/your-working-points-api/stores/postgres/store...)

**Comment by altruity** [comment]
Adding for posterity...

Local terminal should be fine, running the commands below successfully export the db. 
(Note that it doesn't require `psql` in the command string)

When you use the -Fc option with `pg_dump`, it creates a custom-format dump file, which is a binary file. This format is designed to be more efficient and to include additional metadata that makes it suitable for use with `pg_restore`.
`pg_dump -Fc -v --dbname=<vercel_db_source> -f db-example.dump`

If you want a plain text SQL dump that you can view in a text editor like VS Code, you can use the -Fp option instead:
`pg_dump -Fp -v --dbname=<vercel_db_source> -f db-example.sql`

Replace `<vercel_db_source>` with the actual connection string or details provided by Vercel.

Also keep in mind that plain text SQL dumps are generally larger than custom-format dumps because they don't include the same level of compression and optimization. However, they are human-readable and can be easily inspected or modified if needed.
